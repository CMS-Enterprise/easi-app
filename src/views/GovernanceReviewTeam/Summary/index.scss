@use 'uswds-core' as *;
@use 'colors' as *;
@use 'viewports' as *;

.easi-admin-summary {
  &__request-details {
    // Overriding USWDS to change the Breadcrumb caret to white
    // The BG color is blue so by default, there's not enough color contrast
    .usa-breadcrumb__list-item:not(:last-child)::after {
      background-color: color('white');
    }
  }
  
  &__status {
    .grid-row {
      row-gap: .5rem;
    }

    .grid-row > div {
      display: flex;
      align-items: center;
      flex-wrap: wrap;
      column-gap: .5rem;
      row-gap: .25rem;

      h4 {
        margin-top: 0;
        margin-bottom: 0;
      }

      > div {
        display: flex;
        align-items: center;
        width: 100%;

        @media screen and (min-width: $tablet) {
          width: auto;
        }
      }
    }
  }
}
