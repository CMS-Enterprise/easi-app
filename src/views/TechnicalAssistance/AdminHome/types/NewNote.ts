/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TRBAdminNoteCategory } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL fragment: NewNote
// ====================================================

export interface NewNote_author {
  __typename: "UserInfo";
  commonName: string;
}

export interface NewNote {
  __typename: "TRBAdminNote";
  id: UUID;
  isArchived: boolean;
  category: TRBAdminNoteCategory;
  noteText: string;
  author: NewNote_author;
  createdAt: Time;
}
