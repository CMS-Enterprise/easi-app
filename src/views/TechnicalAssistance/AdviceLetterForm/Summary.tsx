import React, { useCallback, useEffect } from 'react';
import { Controller } from 'react-hook-form';
import { Trans, useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { ApolloError, useMutation } from '@apollo/client';
import { yupResolver } from '@hookform/resolvers/yup';
import { ErrorMessage, Form, FormGroup } from '@trussworks/react-uswds';

import HelpText from 'components/shared/HelpText';
import Label from 'components/shared/Label';
import TextAreaField from 'components/shared/TextAreaField';
import useEasiForm from 'hooks/useEasiForm';
import { UpdateTrbAdviceLetterQuery } from 'queries/TrbAdviceLetterQueries';
import {
  UpdateTrbAdviceLetter,
  UpdateTrbAdviceLetterVariables
} from 'queries/types/UpdateTrbAdviceLetter';
import {
  AdviceLetterSummary,
  StepComponentProps
} from 'types/technicalAssistance';
import { meetingSummarySchema } from 'validations/trbRequestSchema';

import { StepSubmit } from '../RequestForm';
import Pager from '../RequestForm/Pager';

const Summary = ({
  trbRequestId,
  adviceLetter,
  setFormAlert,
  setStepSubmit,
  setIsStepSubmitting
}: StepComponentProps) => {
  const { t } = useTranslation('technicalAssistance');
  const history = useHistory();

  const { meetingSummary } = adviceLetter;

  const [update] = useMutation<
    UpdateTrbAdviceLetter,
    UpdateTrbAdviceLetterVariables
  >(UpdateTrbAdviceLetterQuery);

  const {
    handleSubmit,
    control,
    watch,
    partialSubmit,
    formState: { isSubmitting, isDirty }
  } = useEasiForm<AdviceLetterSummary>({
    resolver: yupResolver(meetingSummarySchema),
    defaultValues: {
      meetingSummary
    }
  });

  /** Submit meeting summary fields and update advice letter */
  const submit = useCallback<StepSubmit>(
    (callback, shouldValidate) =>
      handleSubmit(
        async formData => {
          try {
            if (isDirty) {
              // UpdateTrbAdviceLetter mutation
              await update({
                variables: {
                  input: {
                    trbRequestId,
                    ...formData
                  }
                }
              });
            }
            setFormAlert(null);
            callback?.();
          } catch (e) {
            if (e instanceof ApolloError) {
              setFormAlert({
                type: 'error',
                message: t('adviceLetterForm.error', {
                  action: 'saving',
                  type: 'advice letter'
                })
              });
            }
          }
        },
        async () => {
          // Save valid dirty fields on save and exit
          if (!shouldValidate) {
            await partialSubmit({
              update: formData =>
                update({
                  variables: {
                    input: {
                      trbRequestId,
                      ...formData
                    }
                  }
                }),
              callback
            });
          }
        }
      )(),
    [
      handleSubmit,
      isDirty,
      trbRequestId,
      update,
      setFormAlert,
      t,
      partialSubmit
    ]
  );

  useEffect(() => {
    setStepSubmit(() => submit);
  }, [setStepSubmit, submit]);

  useEffect(() => {
    setIsStepSubmitting(isSubmitting);
  }, [setIsStepSubmitting, isSubmitting]);

  return (
    <Form
      onSubmit={e => e.preventDefault()}
      id="trbAdviceSummary"
      className="maxw-tablet"
    >
      {/* Required fields help text */}
      <HelpText className="margin-top-1 margin-bottom-1 text-base">
        <Trans
          i18nKey="technicalAssistance:requiredFields"
          components={{ red: <span className="text-red" /> }}
        />
      </HelpText>

      {/* Meeting summary field */}
      <Controller
        name="meetingSummary"
        control={control}
        render={({ field, fieldState: { error } }) => {
          return (
            <FormGroup error={!!error}>
              <Label className="text-normal" htmlFor="meetingSummary" required>
                {t('adviceLetterForm.meetingSummary')}{' '}
              </Label>
              {error && <ErrorMessage>{t('errors.fillBlank')}</ErrorMessage>}
              <TextAreaField
                id="meetingSummary"
                {...field}
                ref={null}
                value={field.value || ''}
              />
            </FormGroup>
          );
        }}
      />

      {/* Form pager buttons */}
      <Pager
        className="margin-top-4"
        back={{
          outline: true,
          text: t('button.cancel'),
          onClick: () => history.push(`/trb/${trbRequestId}/advice`)
        }}
        next={{
          disabled: isSubmitting || !watch('meetingSummary'),
          onClick: () =>
            submit(() =>
              history.push(`/trb/${trbRequestId}/advice/recommendations`)
            )
        }}
        taskListUrl={`/trb/${trbRequestId}/advice`}
        saveExitText={t('adviceLetterForm.returnToRequest')}
        submit={submit}
        border={false}
      />
    </Form>
  );
};

export default Summary;
