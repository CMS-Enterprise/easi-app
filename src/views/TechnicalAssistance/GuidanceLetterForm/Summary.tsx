import React, { useCallback, useEffect } from 'react';
import { Controller } from 'react-hook-form';
import { Trans, useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { ApolloError, useMutation } from '@apollo/client';
import { yupResolver } from '@hookform/resolvers/yup';
import { ErrorMessage, Form, FormGroup } from '@trussworks/react-uswds';

import useEasiForm from 'components/EasiForm/useEasiForm';
import RichTextEditor from 'components/RichTextEditor';
import HelpText from 'components/shared/HelpText';
import Label from 'components/shared/Label';
import { UpdateTrbGuidanceLetterQuery } from 'queries/TrbAdviceLetterQueries';
import {
  UpdateTrbGuidanceLetter,
  UpdateTrbGuidanceLetterVariables
} from 'queries/types/UpdateTrbGuidanceLetter';
import {
  GuidanceLetterSummary,
  StepComponentProps
} from 'types/technicalAssistance';
import { meetingSummarySchema } from 'validations/trbRequestSchema';

import { StepSubmit } from '../RequestForm';
import Pager from '../RequestForm/Pager';

const Summary = ({
  trbRequestId,
  adviceLetter,
  setFormAlert,
  setStepSubmit,
  setIsStepSubmitting
}: StepComponentProps) => {
  const { t } = useTranslation('technicalAssistance');
  const history = useHistory();

  const { meetingSummary } = adviceLetter;

  const [update] = useMutation<
    UpdateTrbGuidanceLetter,
    UpdateTrbGuidanceLetterVariables
  >(UpdateTrbGuidanceLetterQuery);

  const {
    handleSubmit,
    control,
    watch,
    partialSubmit,
    formState: { isSubmitting, isDirty }
  } = useEasiForm<GuidanceLetterSummary>({
    resolver: yupResolver(meetingSummarySchema),
    defaultValues: {
      meetingSummary
    }
  });

  /** Submit meeting summary fields and update guidance letter */
  const submit = useCallback<StepSubmit>(
    (callback, shouldValidate = true) =>
      handleSubmit(
        async formData => {
          try {
            if (isDirty) {
              // UpdateTrbAdviceLetter mutation
              await update({
                variables: {
                  input: {
                    trbRequestId,
                    ...formData
                  }
                }
              });
            }
            setFormAlert(null);
            callback?.();
          } catch (e) {
            if (e instanceof ApolloError) {
              setFormAlert({
                type: 'error',
                message: t('guidanceLetterForm.error', {
                  action: 'saving',
                  type: 'guidance letter'
                })
              });
            }
          }
        },
        async () => {
          // Save valid dirty fields on save and exit
          if (!shouldValidate) {
            await partialSubmit({
              update: formData =>
                update({
                  variables: {
                    input: {
                      trbRequestId,
                      ...formData
                    }
                  }
                }),
              callback
            });
          }
        }
      )(),
    [
      handleSubmit,
      isDirty,
      trbRequestId,
      update,
      setFormAlert,
      t,
      partialSubmit
    ]
  );

  useEffect(() => {
    setStepSubmit(() => submit);
  }, [setStepSubmit, submit]);

  useEffect(() => {
    setIsStepSubmitting(isSubmitting);
  }, [setIsStepSubmitting, isSubmitting]);

  return (
    <Form
      onSubmit={e => e.preventDefault()}
      id="trbGuidanceSummary"
      className="maxw-tablet"
    >
      {/* Required fields help text */}
      <HelpText className="margin-top-1 margin-bottom-1 text-base">
        <Trans
          i18nKey="technicalAssistance:requiredFields"
          components={{ red: <span className="text-red" /> }}
        />
      </HelpText>

      {/* Meeting summary field */}

      <Controller
        name="meetingSummary"
        control={control}
        render={({ field, fieldState: { error } }) => {
          return (
            <FormGroup error={!!error}>
              <Label
                className="text-normal"
                id="meetingSummary-label"
                htmlFor="meetingSummary"
                required
              >
                {t('guidanceLetterForm.meetingSummary')}{' '}
              </Label>
              {error && <ErrorMessage>{t('errors.fillBlank')}</ErrorMessage>}
              <RichTextEditor
                editableProps={{
                  id: 'meetingSummary',
                  'data-testid': 'meetingSummary',
                  'aria-describedby': 'meetingSummary-hint',
                  'aria-labelledby': 'meetingSummary-label'
                }}
                field={{ ...field, value: field.value || '' }}
                required
              />
            </FormGroup>
          );
        }}
      />

      {/* Form pager buttons */}
      <Pager
        className="margin-top-4"
        back={{
          outline: true,
          text: t('button.cancel'),
          onClick: () => history.push(`/trb/${trbRequestId}/guidance`)
        }}
        next={{
          disabled: isSubmitting || !watch('meetingSummary'),
          onClick: () =>
            submit(() => history.push(`/trb/${trbRequestId}/guidance/insights`))
        }}
        taskListUrl={`/trb/${trbRequestId}/guidance`}
        saveExitText={t('guidanceLetterForm.returnToRequest')}
        submit={submit}
        border={false}
      />
    </Form>
  );
};

export default Summary;
