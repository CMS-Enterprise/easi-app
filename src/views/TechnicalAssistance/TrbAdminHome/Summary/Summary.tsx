import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Grid,
  GridContainer
} from '@trussworks/react-uswds';

type SummaryProps = {
  trbRequestId: string;
  name: string;
  requestType: string;
  requester: string;
  submissionDate: string;
};

export default function Summary({
  trbRequestId,
  name,
  requestType,
  requester,
  submissionDate
}: SummaryProps) {
  const { t } = useTranslation('technicalAssistance');
  return (
    <section className="bg-primary-darker padding-bottom-3 text-white">
      <GridContainer>
        {/* Breadcrumbs */}
        <BreadcrumbBar variant="wrap" className="bg-transparent text-white">
          <Breadcrumb>
            <BreadcrumbLink asCustom={Link} to="/">
              <span className="text-white">{t('Home')}</span>
            </BreadcrumbLink>
          </Breadcrumb>
          <Breadcrumb current>
            {t('adminHome.breadcrumb', { trbRequestId })}
          </Breadcrumb>
        </BreadcrumbBar>

        {/* Request name */}
        <h2 className="margin-top-05 margin-bottom-0">{name}</h2>

        {/* Request details */}
        <Grid row>
          <Grid col tablet={{ col: 6 }}>
            {/* Request type */}
            <h5 className="text-normal margin-bottom-05 margin-top-2">
              {t('adminHome.requestType')}
            </h5>
            <h4 className="margin-y-05">{t(requestType)}</h4>
          </Grid>
          <Grid col tablet={{ col: 6 }}>
            {/* Requester */}
            <h5 className="text-normal margin-bottom-05 margin-top-2">
              {t('adminHome.requester')}
            </h5>
            <h4 className="margin-y-05">{t(requester)}</h4>
            {/* Submission date */}
            <h5 className="text-normal margin-bottom-05 margin-top-2">
              {t('adminHome.submissionDate')}
            </h5>
            <h4 className="margin-y-05">{t(submissionDate)}</h4>
          </Grid>
        </Grid>
      </GridContainer>
    </section>
  );
}
