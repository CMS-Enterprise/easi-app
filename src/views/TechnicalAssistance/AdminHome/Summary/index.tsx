import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  Grid,
  GridContainer,
  IconError
} from '@trussworks/react-uswds';
import { DateTime } from 'luxon';

import cmsDivisionsAndOffices from 'constants/enums/cmsDivisionsAndOffices';
import useTRBAttendees from 'hooks/useTRBAttendees';
import { GetTrbRequest_trbRequest_taskStatuses as TRBRequestTaskStatuses } from 'queries/types/GetTrbRequest';
import { TRBRequestStatus, TRBRequestType } from 'types/graphql-global-types';

type SummaryProps = {
  trbRequestId: string;
  name: string;
  requestType: TRBRequestType;
  createdAt: string;
  createdBy: string;
  status: TRBRequestStatus;
  taskStatuses: TRBRequestTaskStatuses;
  trbLead: string | null;
};

export default function Summary({
  trbRequestId,
  name,
  requestType,
  createdAt,
  createdBy,
  status,
  taskStatuses,
  trbLead
}: SummaryProps) {
  const { t } = useTranslation('technicalAssistance');

  /**
   * Request submission date for summary
   */
  const submissionDate = createdAt
    ? DateTime.fromISO(createdAt).toLocaleString(DateTime.DATE_FULL)
    : '';

  // Get requester object from request attendees
  const {
    data: { requester, loading }
  } = useTRBAttendees(trbRequestId, createdBy);

  /**
   * Requester info for display in summary box
   */
  const requesterString = useMemo(() => {
    // If loading, return null
    if (loading) return null;

    // If requester component is not set, return name or EUA
    if (!requester.component)
      return requester?.userInfo?.commonName || requester?.userInfo?.euaUserId;

    /** Requester component */
    const requesterComponent = cmsDivisionsAndOffices.find(
      object => object.name === requester.component
    );

    // Return requester name and component acronym
    return `${requester?.userInfo?.commonName}, ${requesterComponent?.acronym}`;
  }, [requester, loading]);

  /** Get current task status */
  const currentTaskStatus: keyof TRBRequestTaskStatuses = useMemo(() => {
    /** Task status object keys */
    const statusKeys = [
      'formStatus',
      'feedbackStatus',
      'attendConsultPrepStatus',
      'consultPrepStatus'
    ] as (keyof TRBRequestTaskStatuses)[];

    /** Current step in the TRB request task list */
    // Finds first task status that is not completed
    // Returns undefined if all steps completed
    const currentStatus = statusKeys.find(
      statusKey => taskStatuses[statusKey] !== 'COMPLETED'
    );

    // Return current status
    // If all task list steps have been completed, return attendConsultStatus
    return currentStatus || 'attendConsultStatus';
  }, [taskStatuses]);

  /** Corresponding task status text from translation file */
  const taskStatusText: string = t(
    `adminHome.taskStatuses.${currentTaskStatus}.${taskStatuses[currentTaskStatus]}`
  );

  return (
    <div className="trb-admin-home__summary">
      <section className="bg-primary-darker padding-bottom-3 text-white">
        <GridContainer>
          {/* Breadcrumbs */}
          <BreadcrumbBar variant="wrap" className="bg-transparent text-white">
            <Breadcrumb>
              <BreadcrumbLink asCustom={Link} to="/">
                <span className="text-white">{t('Home')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb current>
              {t('adminHome.breadcrumb', { trbRequestId })}
            </Breadcrumb>
          </BreadcrumbBar>

          {/* Request name */}
          <h2 className="margin-top-05 margin-bottom-0">{name}</h2>

          {/* Request details */}
          <Grid row>
            <Grid tablet={{ col: 8 }}>
              {/* Request type */}
              <div
                data-testid="trbSummary-requestType"
                className="margin-top-2 margin-bottom-05"
              >
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.requestType')}
                </h5>
                <h4 className="margin-y-05">
                  {t(`requestType.type.${requestType}.heading`)}
                </h4>
              </div>
            </Grid>
            <Grid tablet={{ col: 4 }}>
              {/* Requester */}
              <div className="margin-top-2 margin-bottom-05">
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.requester')}
                </h5>
                {requesterString && (
                  <h4
                    className="margin-y-05"
                    data-testid={`trbSummary-requester_${requester.userInfo?.euaUserId}`}
                  >
                    {requesterString}
                  </h4>
                )}
              </div>
              {/* Submission date */}
              <div
                data-testid="trbSummary-submissionDate"
                className="margin-top-2 margin-bottom-05"
              >
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.submissionDate')}
                </h5>
                <h4 className="margin-y-05">{t(submissionDate)}</h4>
              </div>
            </Grid>
          </Grid>
        </GridContainer>
      </section>

      {/* Status bar */}
      <section className="bg-base-lightest padding-y-05">
        <GridContainer>
          <Grid row>
            {/* Status */}
            <Grid
              tablet={{ col: 8 }}
              data-testid="trbSummary-status"
              className="display-flex flex-align-center margin-y-05"
            >
              <h4 className="margin-y-0">{t('adminHome.status')}</h4>
              <span className="bg-info-dark text-white text-bold padding-y-05 padding-x-105 margin-x-1">
                {t(`adminHome.${status.toLowerCase()}`)}
              </span>
              <p className="margin-y-0 text-base">{taskStatusText}</p>
            </Grid>

            {/* TRB Lead */}
            <Grid
              tablet={{ col: 4 }}
              data-testid="trbSummary-trbLead"
              className="display-flex flex-align-center margin-y-05"
            >
              <h4 className="margin-y-0">{t('adminHome.trbLead')}</h4>
              <p className="margin-y-0 margin-x-1 display-flex flex-align-center">
                {
                  // Display trb lead name or not assigned
                  trbLead || (
                    <>
                      <IconError className="text-error margin-right-05" />
                      {t('adminHome.notAssigned')}
                    </>
                  )
                }
              </p>
              <Button unstyled type="button" className="width-auto">
                {trbLead ? t('adminHome.change') : t('adminHome.assign')}
              </Button>
            </Grid>
          </Grid>
        </GridContainer>
      </section>
    </div>
  );
}
