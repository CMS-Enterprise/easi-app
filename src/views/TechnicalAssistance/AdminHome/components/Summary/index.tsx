import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Grid,
  GridContainer,
  IconError,
  ModalRef
} from '@trussworks/react-uswds';
import classNames from 'classnames';

import { TRBAttendee } from 'queries/types/TRBAttendee';
import {
  TRBRequestState,
  TRBRequestStatus,
  TRBRequestType
} from 'types/graphql-global-types';
import { TrbRequestIdRef } from 'types/technicalAssistance';

import { TrbAssignLeadModalOpener } from '../../TrbAssignLeadModal';

type SummaryProps = {
  trbRequestId: string;
  name: string | null;
  requestType: TRBRequestType;
  state: TRBRequestState;
  taskStatus?: TRBRequestStatus;
  trbLead: string | null;
  requester: TRBAttendee;
  requesterString?: string | null;
  submissionDate: string;
  assignLeadModalRef: React.RefObject<ModalRef>;
  assignLeadModalTrbRequestIdRef: React.MutableRefObject<TrbRequestIdRef>;
};

export default function Summary({
  trbRequestId,
  name,
  requestType,
  state,
  taskStatus,
  trbLead,
  requester,
  requesterString,
  submissionDate,
  assignLeadModalRef,
  assignLeadModalTrbRequestIdRef
}: SummaryProps) {
  const { t } = useTranslation('technicalAssistance');

  const taskStatusText: string = t(
    `adminHome.trbRequestStatuses.${taskStatus}`
  );

  return (
    <div className="trb-admin-home__summary">
      <section className="bg-primary-darker padding-bottom-3 text-white">
        <GridContainer>
          {/* Breadcrumbs */}
          <BreadcrumbBar variant="wrap" className="bg-transparent text-white">
            <Breadcrumb>
              <BreadcrumbLink asCustom={Link} to="/">
                <span className="text-white">{t('adminHome.home')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb current>{t('adminHome.breadcrumb')}</Breadcrumb>
          </BreadcrumbBar>

          {/* Request name */}
          <h2 className="margin-top-05 margin-bottom-0">{name}</h2>

          {/* Request details */}
          <Grid row>
            <Grid tablet={{ col: 8 }}>
              {/* Request type */}
              <div
                data-testid="trbSummary-requestType"
                className="margin-top-2 margin-bottom-05"
              >
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.requestType')}
                </h5>
                <h4 className="margin-y-05">
                  {t(`requestType.type.${requestType}.heading`)}
                </h4>
              </div>
            </Grid>
            <Grid tablet={{ col: 4 }}>
              {/* Requester */}
              <div className="margin-top-2 margin-bottom-05">
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.requester')}
                </h5>
                {requesterString && (
                  <h4
                    className="margin-y-05"
                    data-testid={`trbSummary-requester_${requester.userInfo?.euaUserId}`}
                  >
                    {requesterString}
                  </h4>
                )}
              </div>
              {/* Submission date */}
              <div
                data-testid="trbSummary-submissionDate"
                className="margin-top-2 margin-bottom-05"
              >
                <h5 className="text-normal margin-y-0">
                  {t('adminHome.submissionDate')}
                </h5>
                <h4 className="margin-y-05">{t(submissionDate)}</h4>
              </div>
            </Grid>
          </Grid>
        </GridContainer>
      </section>

      {/* Status bar */}
      <section className="bg-base-lightest padding-y-05">
        <GridContainer>
          <Grid row>
            {/* Status */}
            <Grid
              tablet={{ col: 8 }}
              data-testid="trbSummary-status"
              className="display-flex flex-align-center margin-y-05"
            >
              <h4 className="margin-y-0">{t('adminHome.status')}</h4>
              <span
                className={classNames(
                  'text-white text-bold padding-y-05 padding-x-105 margin-x-1',
                  {
                    'bg-base': state === TRBRequestState.CLOSED,
                    'bg-info-dark': state === TRBRequestState.OPEN
                  }
                )}
              >
                {t(`adminHome.${state.toLowerCase()}`)}
              </span>
              <p className="margin-y-0 text-base">{taskStatusText}</p>
            </Grid>

            {/* TRB Lead */}
            <Grid
              tablet={{ col: 4 }}
              data-testid="trbSummary-trbLead"
              className="display-flex flex-align-center margin-y-05"
            >
              <h4 className="margin-y-0">{t('adminHome.trbLead')}</h4>
              <p className="margin-y-0 margin-x-1 display-flex flex-align-center">
                {
                  // Display trb lead name or not assigned
                  trbLead || (
                    <>
                      <IconError className="text-error margin-right-05" />
                      {t('adminHome.notAssigned')}
                    </>
                  )
                }
              </p>
              <TrbAssignLeadModalOpener
                trbRequestId={trbRequestId}
                modalRef={assignLeadModalRef}
                trbRequestIdRef={assignLeadModalTrbRequestIdRef}
                className="usa-button--unstyled width-auto"
              >
                {trbLead ? t('adminHome.change') : t('adminHome.assign')}
              </TrbAssignLeadModalOpener>
            </Grid>
          </Grid>
        </GridContainer>
      </section>
    </div>
  );
}
