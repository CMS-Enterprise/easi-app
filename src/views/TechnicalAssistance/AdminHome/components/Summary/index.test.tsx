import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, within } from '@testing-library/react';
import { ModalRef } from '@trussworks/react-uswds';

import {
  getTRBRequestAttendeesQuery,
  getTrbRequestQuery,
  requester,
  trbRequestSummary
} from 'data/mock/trbRequest';
import { TrbRequestIdRef } from 'types/technicalAssistance';
import easiMockStore from 'utils/testing/easiMockStore';
import { mockTrbRequestId } from 'utils/testing/MockTrbAttendees';

import Summary from '.';

const defaultStore = easiMockStore({
  euaUserId: 'SF13',
  groups: ['EASI_TRB_ADMIN_D']
});

describe('TRB Admin Home summary', () => {
  const modalRef = React.createRef<ModalRef>();
  const trbRequestIdRef = React.createRef<TrbRequestIdRef>();

  it('renders TRB request details', async () => {
    const { asFragment, findByTestId, getByTestId } = render(
      <MemoryRouter>
        <MockedProvider
          mocks={[getTrbRequestQuery, getTRBRequestAttendeesQuery]}
        >
          <Provider store={defaultStore}>
            <Summary
              trbRequestId={mockTrbRequestId}
              name={trbRequestSummary.name || 'Draft'}
              requestType={trbRequestSummary.type}
              state={trbRequestSummary.state}
              taskStatus={trbRequestSummary.status}
              trbLead={trbRequestSummary.trbLeadInfo.commonName}
              requester={requester}
              requesterString="Adeline Aarons, CMS"
              submissionDate="January 5, 2023"
              assignLeadModalRef={modalRef}
              assignLeadModalTrbRequestIdRef={trbRequestIdRef}
            />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    /** Formatted requester string */
    const requesterString = await findByTestId(
      `trbSummary-requester_${requester?.userInfo?.euaUserId}`
    );

    // Check that requester shows correct name and component acronym
    expect(requesterString).toHaveTextContent('Adeline Aarons, CMS');

    // Check that submission date is formatted correctly
    expect(getByTestId('trbSummary-submissionDate')).toHaveTextContent(
      'January 5, 2023'
    );

    // Check that correct task status is rendered
    expect(getByTestId('trbSummary-status')).toHaveTextContent(
      'Draft request form'
    );

    /** TRB Lead container */
    const trbLeadContainer = getByTestId('trbSummary-trbLead');

    // Check that TRB lead shows as not assigned
    expect(within(trbLeadContainer).getByText('Not assigned'));

    // Check that assign button is rendered
    expect(within(trbLeadContainer).getByRole('button', { name: 'Assign' }));

    // Snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
