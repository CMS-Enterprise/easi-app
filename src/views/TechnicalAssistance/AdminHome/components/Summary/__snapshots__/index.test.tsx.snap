// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TRB Admin Home summary renders TRB request details 1`] = `
<DocumentFragment>
  <div
    class="trb-admin-home__summary"
  >
    <section
      class="bg-primary-darker padding-bottom-3 text-white"
    >
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <nav
          aria-label="Breadcrumbs"
          class="usa-breadcrumb usa-breadcrumb--wrap bg-transparent text-white"
        >
          <ol
            class="usa-breadcrumb__list"
          >
            <li
              class="usa-breadcrumb__list-item"
            >
              <a
                class="usa-breadcrumb__link"
                href="/"
              >
                <span
                  class="text-white"
                >
                  Home
                </span>
              </a>
            </li>
            <li
              aria-current="page"
              class="usa-breadcrumb__list-item usa-current"
            >
              Request details
            </li>
          </ol>
        </nav>
        <h2
          class="margin-top-05 margin-bottom-0"
        >
          Mock TRB Request
        </h2>
        <div
          class="grid-row"
          data-testid="grid"
        >
          <div
            class="tablet:grid-col-8"
            data-testid="grid"
          >
            <div
              class="margin-top-2 margin-bottom-05"
              data-testid="trbSummary-requestType"
            >
              <h5
                class="text-normal margin-y-0"
              >
                Request type
              </h5>
              <h4
                class="margin-y-05"
              >
                Iâ€™m having a problem with my system
              </h4>
            </div>
          </div>
          <div
            class="tablet:grid-col-4"
            data-testid="grid"
          >
            <div
              class="margin-top-2 margin-bottom-05"
            >
              <h5
                class="text-normal margin-y-0"
              >
                Requester
              </h5>
              <h4
                class="margin-y-05"
                data-testid="trbSummary-requester_ABCD"
              >
                Adeline Aarons, CMS
              </h4>
            </div>
            <div
              class="margin-top-2 margin-bottom-05"
              data-testid="trbSummary-submissionDate"
            >
              <h5
                class="text-normal margin-y-0"
              >
                Submission Date
              </h5>
              <h4
                class="margin-y-05"
              >
                January 5, 2023
              </h4>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="bg-base-lightest padding-y-05"
    >
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          class="grid-row"
          data-testid="grid"
        >
          <div
            class="tablet:grid-col-8 display-flex flex-align-center margin-y-05"
            data-testid="trbSummary-status"
          >
            <h4
              class="margin-y-0"
            >
              Status
            </h4>
            <span
              class="text-white text-bold padding-y-05 padding-x-105 margin-x-1 bg-info-dark"
            >
              Open
            </span>
            <p
              class="margin-y-0 text-base"
            >
              Draft request form
            </p>
          </div>
          <div
            class="tablet:grid-col-4 display-flex flex-align-center margin-y-05"
            data-testid="trbSummary-trbLead"
          >
            <h4
              class="margin-y-0"
            >
              TRB Lead
            </h4>
            <p
              class="margin-y-0 margin-x-1 display-flex flex-align-center"
            >
              <svg
                class="usa-icon text-error margin-right-05"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
                />
              </svg>
              Not assigned
            </p>
            <button
              class="usa-button usa-button--unstyled width-auto"
              type="button"
            >
              Assign
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
